app/components/Layout/Header.tsx

```diff
@@ -80,19 +92,64 @@ const HeaderLayout = ({
             {isLoggedIn ? (
               <>
                 {isRegistered && (
-                  <li>
-                    <Link href="/user/profile">
-                      <a className="nav-button">Profile</a>
-                    </Link>
-                  </li>
+                  <>
+                    <li className="d-none d-lg-block">
+                      <Dropdown alignRight>
+                        <Dropdown.Toggle variant="primary">
+                          <FontAwesomeIcon
+                            color="white"
+                            icon={faUser}
+                            size="1x"
+                          />
+                        </Dropdown.Toggle>
+                        <Dropdown.Menu>
+                          <Dropdown.Item as={Link} href="/user/profile">
+                            <a className="dropdown-item text-right">
+                              <div>
+                                <span>{user.firstName}</span>{' '}
+                                <span>{user.lastName}</span>
+                              </div>
+                              <div className="small text-muted text-nowrap">
+                                {user.emailAddress}
+                              </div>
+                            </a>
+                          </Dropdown.Item>
+                          <Dropdown.Item
+                            as={Form}
+                            action="/logout"
+                            method="post"
+                            tabIndex={0}
+                            onKeyPress={submitForm}
```

Comment:
The HTML form appears to submit fine for me without an imperative handler - was there some issue with form submission?

Reply:
I was trying keyboard navigation on the new dropdown, but it didn't submit without this.